sdk/azidentity/azidentity.go

```diff
@@ -151,6 +151,14 @@ func doForClient(client *azcore.Client, r *http.Request) (*http.Response, error)
 			return nil, err
 		}
 	}
+
+	h := req.Raw().Header
```

Comment:
I have marked the commit as `fix`, because, at first glance, this looks like an oversight, but there might be a clear reason for not propagating the headers further - I just haven't been able to figure it out and the PR that added it (https://github.com/Azure/azure-sdk-for-go/pull/16243) isn't very descriptive in this regard

in any case, feel free to elaborate on the intentions behind this method

Reply:
Looks like an oversight to me, I don't know a reason to drop the incoming request's headers. I think azidentity should propagate headers that don't conflict with the ones it sets, to avoid breaking scenarios like yours. The incoming headers are an MSAL implementation detail azidentity should avoid having an opinion about. Any thoughts @jhendrixMSFT?

Reply:
Agree 100%.